### PR TITLE
Added namespaces and fixed various issues with language and code

### PR DESCRIPTION
**Pull request by ellistarn**

### CODE REVIEW

**Review by njtran — CHANGES_REQUESTED** [review]
Good work! Looks really nice and a lot cleaner to do.

File: queue-length/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            - -c
+            - >
+              while true; do
+                aws sqs delete-message --queue-url ${QUEUE_URL} --receipt-handle $(aws sqs receive-message --queue-url ${QUEUE_URL} | jq -r '.Messages[0].ReceiptHandle')
+                sleep 30;
+              done
+          resources:
+            requests:
+              memory: 100Mi
+              cpu: 1
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply queue-length/manifest.yaml:60]: Do you mind doing this in a followon so you can test it at the same time?

File: queue-length/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+spec:
+  scaleTargetRef:
+    apiVersion: autoscaling.karpenter.sh/v1alpha1
+    kind: ScalableNodeGroup
+    name: capacity
+  minReplicas: 1
+  maxReplicas: 10
+  behavior:
+    scaleDown:
+      stabilizationWindowSeconds: 60
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply queue-length/manifest.yaml:76]: Yeah I was seeing some oscillation with this setting (churn up/down). But I think it needs to be fast for the demo, as long as it doesn't bottom out, it doesn't look weird.

File: queue-length/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: karpenter-queue-length-demo
+---
+apiVersion: autoscaling.karpenter.sh/v1alpha1
+kind: MetricsProducer
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply queue-length/manifest.yaml:7]: Let's just add it here if you want to include it for your demo. You can decorate it with comments to let users know its only for visualization.

File: queue-length/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Queue Length Demo
+
+This demo will use the length of an AWS SQS Queue to autoscale a Deployment and Node Group.
+
+We will deploy pods that handle 1 message every 30 seconds. The autoscalers are configured to create 4 nodes per pod.
+
+## Apply YAML and Watch
+
+```bash
+wget https://raw.githubusercontent.com/ellistarn/karpenter-aws-demo/main/queue/manifest.yaml
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply queue-length/README.md:10]: :!

File: queue-length/manifest.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (njtran) [reply queue-length/manifest.yaml:60]: Sure. Can do.

File: queue-length/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ellistarn) [reply queue-length/README.md:10]: You reminded me to change the other one to wget https://raw.githubusercontent.com/ellistarn/karpenter-aws-demo/main/queue-length/manifest.yaml

File: queue-length/manifest.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (njtran) [reply queue-length/manifest.yaml:76]: As long as we're keeping it as 1 message/sec I think this is good.

File: queue-length/manifest.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (njtran) [reply queue-length/manifest.yaml:7]: Want me to do it in a follow up? Or are you doing it?

**Review by njtran — APPROVED** [review]
LGTM!

**Pull request merged**